src/frontend/apps/markdown/MarkdownEditor/index.spec.tsx

```diff
@@ -0,0 +1,321 @@
+    fetchMock.get('/api/markdown-documents/1/', markdownDocument);
+
+    render(
+      wrapInIntlProvider(
+        <QueryClientProvider client={queryClient}>
+          <MarkdownEditor />
+        </QueryClientProvider>,
+      ),
+    );
+
+    // Wait for rendered content
+    await waitFor(() =>
+      expect(
+        screen.getAllByText(markdownDocument.translations[0].content).length, // en translation
+      ).toEqual(2),
+    );
+
+    // Display editor only
+    userEvent.click(screen.getByTestId('disposition-editor-only'));
+    expect(screen.getByTestId('editor_container')).toBeVisible();
+    expect(screen.getByTestId('renderer_container')).not.toBeVisible();
+
+    // Display rendering only
+    userEvent.click(screen.getByTestId('disposition-rendering-only'));
+    expect(screen.getByTestId('editor_container')).not.toBeVisible();
+    expect(screen.getByTestId('renderer_container')).toBeVisible();
+
+    // Display both again
+    userEvent.click(screen.getByTestId('disposition-split-screen'));
+    expect(screen.getByTestId('editor_container')).toBeVisible();
+    expect(screen.getByTestId('renderer_container')).toBeVisible();
+  });
+
+  it('enable MDX', async () => {
+    const mdxContent =
+      '# MDX\n' +
+      '\n' +
+      'This paragraph shows what are the advanced uses provided by using *MDX*\n' +
+      '\n' +
+      "<div style={{padding: '1rem', backgroundColor: 'violet'}} data-testid=\"mdx-rendered-block\">\n" +
+      '  Instructor can override the design of the document.\n' +
+      '</div>\n' +
+      '\n' +
+      '\n' +
+      'One can define variables:\n' +
+      '\n' +
+      'export const good_decade = "70th"\n' +
+      '\n' +
+      'And use it in text like in the {good_decade}';
+
+    const markdownDocument = markdownDocumentMockFactory({
+      id: '1',
+      is_draft: true,
+      translations: [
+        markdownTranslationMockFactory({
+          language_code: 'en',
+          content: mdxContent,
+        }),
+      ],
+    });
+    const queryClient = new QueryClient();
+    fetchMock.get('/api/markdown-documents/1/', markdownDocument);
+
+    render(
+      wrapInIntlProvider(
+        <QueryClientProvider client={queryClient}>
+          <MarkdownEditor />
+        </QueryClientProvider>,
+      ),
+    );
+
+    // Wait for rendered content
+    await waitFor(() =>
+      expect(
+        screen.getByRole('heading', { level: 1, name: 'MDX' }),
+      ).toBeInTheDocument(),
+    );
+
+    // Assert the MDX is not rendered properly
+    expect(
+      within(screen.getByTestId('renderer_container')).getByText(
+        'export const good_decade = "70th"',
+      ),
+    ).toBeInTheDocument();
+    expect(screen.queryByTestId('mdx-rendered-block')).not.toBeInTheDocument();
+
+    // Enable MDX rendering
+    userEvent.click(screen.getByRole('button', { name: 'Settings' })); // open the settings dropdown
+
+    act(() => {
+      userEvent.click(screen.getByTestId('mdx-options-toggle-mdx'));
```

Comment:
```suggestion
      userEvent.click(screen.getByRole('checkbox, { name: 'MDX disabled' }));
```

You can remove the usage of `getByTestId` here because there is an avaiable role you can use.